sled-diagnostics/src/logs.rs

```diff
@@ -942,36 +965,41 @@ mod illumos_tests {
             let mut log_snapshots = LogSnapshots::new();
 
             // Create a new snapshot
-            log_snapshots.get_or_create(&log, &mountpoint).unwrap();
-            let snapshots = get_sled_diagnostics_snapshots(zfs_filesystem);
+            log_snapshots.get_or_create(&log, &mountpoint).await.unwrap();
+            let snapshots =
+                get_sled_diagnostics_snapshots(zfs_filesystem).await;
             assert_eq!(snapshots.len(), 1, "single snapshot created");
 
             // Creating a second snapshot from the same dataset doesn't create a
             // new snapshot
-            log_snapshots.get_or_create(&log, &mountpoint).unwrap();
-            let snapshots = get_sled_diagnostics_snapshots(zfs_filesystem);
+            log_snapshots.get_or_create(&log, &mountpoint).await.unwrap();
+            let snapshots =
+                get_sled_diagnostics_snapshots(zfs_filesystem).await;
             assert_eq!(snapshots.len(), 1, "duplicate snapshots not taken");
 
             // Free all of the log_snapshots
             drop(log_snapshots);
 
-            let snapshots = get_sled_diagnostics_snapshots(zfs_filesystem);
+            let snapshots =
+                get_sled_diagnostics_snapshots(zfs_filesystem).await;
             assert!(snapshots.is_empty(), "no snapshots left behind");
```

Comment:
I don't think it's critical that the drop method is synchronous however it does introduce a race condition in the test here, we may be able to work around it.

Reply:
I've changed the behavior here to use a more explicit drop, and log errors if this wasn't handled. I also tried to refactor above to make this case less likely -- cleaning up snapshots at a higher-level in the call to `get_zone_logs`